src/Symfony/Component/Inflector/Inflector.php

```diff
@@ -218,8 +227,14 @@ public static function singularize($plural)
             }
         }
 
+        $irregularNouns = array(
+            'committee' => 'committee',
+            'feedback' => 'feedback',
+            'screenshots' => 'screenshot',
+        );
+
         // Convert teeth to tooth, feet to foot
-        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3 && 'feedback' !== $plural) {
+        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3 && !isset(self::$irregularNouns[strtolower($plural)])) {
```

Comment:
Can't we take a different approach here? Whitelist instead of blacklist?

It's still problematic, e.g. feeds => foods